python_scripts/cross_validation_sol_02.py

```diff
@@ -135,6 +135,37 @@
 # We see that using `strategy="stratified"`, the results are much worse than
 # with the `most_frequent` strategy. Since the classes are imbalanced,
 # predicting the most frequent involves that we will be right for the
-# proportion of this class (~75% of the samples). However, by using the
-# `stratified` strategy, wrong predictions will be made even for the most
+# proportion of this class (~75% of the samples). However, the `stratified`
+# strategy will randomly generate predictions by respecting the training
+# set’s class distribution, resulting in wrong predictions even for the most
 # frequent class, hence we obtain a lower accuracy.
+#
+# Notice that the `most_frequent` strategy is deterministic for a given
+# cross-validation split, whereas the `stratified` strategy will lead
+# to different results even for the same cv split due to the extra randomization.
+
+# %% tags=["solution"]
```

Comment:
I am not sure that we need to add an example. We can link to the documentation of the `random_state` parameter that documents the random selection using `"uniform"` and `"stratify"`. I don't think that the `DummyClassifier` being non-deterministic is actually one of the messages to convey in the lecture. A note bringing a bit more insight would suffice, IMO.

Reply:
I agree. I think we can keep things simple here. Maybe we could even not mention `strategy="stratify"` at all and simplify the exercise to remove the last question.

In my opinion, what we want to convey with this exercise is that it's useful to always compare your classifier performance to `DummyClassifier(strategy="most_frequent")` as a baseline classifier with a prediction function that totally ignores `X_train` and `X_test` yet can reach seemingly high accuracy scores on imbalanced problems.

The other strategies are only useful for more advanced users who might be curious about other ways to define baselines but they are often useless for practitioners compared to `strategy="most_frequent"` unless one is interested in theoretical considerations that are beyond the target audience of the MOOC.